src/Features/CSharp/Portable/ConvertForToForEach/CSharpConvertForToForEachCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Threading;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.ConvertForToForEach;
+using Microsoft.CodeAnalysis.CSharp.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle.TypeStyle;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertForToForEach
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(CSharpConvertForToForEachCodeRefactoringProvider)), Shared]
+    internal class CSharpConvertForToForEachCodeRefactoringProvider :
+        AbstractConvertForToForEachCodeRefactoringProvider<
+            StatementSyntax, 
+            ForStatementSyntax, 
+            ExpressionSyntax,
+            MemberAccessExpressionSyntax,
+            TypeSyntax,
+            VariableDeclaratorSyntax>
+    {
+        protected override string GetTitle()
+            => CSharpFeaturesResources.Convert_for_to_foreach;
```

Comment:
I see so, you actually created 2 different resources. okay I can do same.

Reply:
Yup yup.